de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/location/geocoding/openstreetmap/NominatimOSMGeoCoordinatesProviderImpl.java

```diff
@@ -0,0 +1,128 @@
+package org.adempiere.location.geocoding.openstreetmap;
+
+import com.google.common.annotations.VisibleForTesting;
+import de.metas.util.GuavaCollectors;
+import lombok.NonNull;
+import org.adempiere.location.geocoding.GeoCoordinatesProvider;
+import org.adempiere.location.geocoding.GeographicalCoordinates;
+import org.adempiere.location.geocoding.GeoCoordinatesRequest;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/*
+ * #%L
+ * metasfresh-pharma
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+@Component
+public class NominatimOSMGeoCoordinatesProviderImpl implements GeoCoordinatesProvider
+{
+
+	/**
+	 * Please observe the 2 URLs.
+	 * Issue with WRONG_BASE_URL is that when using <pre>search?q={address}&postalcode={postalcode}</pre>
+	 * Nominatim tries to search for BOTH a postal and an address. <br/>
+	 *
+	 * In some cases the address may be wrong so we get no response. <br>
+	 *
+	 * With the second BASE_URL Nominatim ignores the address and just searches for postal.
+	 *
+	 * Country is taken into account in both cases.
+	 */
+	//	private static final String WRONG_BASE_URL = "https://nominatim.openstreetmap.org/search?q={address}&postalcode={postalcode}&countrycodes={countrycodes}&format={format}&dedupe={dedupe}&email={email}&polygon_geojson={polygon_geojson}&polygon_kml={polygon_kml}&polygon_svg={polygon_svg}&polygon_text={polygon_text}";
```

Comment:
should we clean this up?

Reply:
I didn't want to clean that up as the difference between the 2 URLs is important to notice. 

One works (ignores bad addresses) while the other doesn't, just because they are constructed differently.